### PR TITLE
Update homepage: November 8, 2018

### LINKED ISSUE #6790: 8 November homepage programming update
For the **8 November** push, please update the homepage to reflect the programming detailed [here](https://docs.google.com/document/d/1g4RuGKW__eTHNd277rKPMM46ygonXzW68jSs2OXEkPM/edit#). 

@muffinresearch @bobsilverberg

### PR DESCRIPTION
**Pull request by willdurand**
Fixes #6790

---

See this Google doc: https://docs.google.com/document/d/1g4RuGKW__eTHNd277rKPMM46ygonXzW68jSs2OXEkPM/edit#

ℹ️ there is more code that expected because I decided to keep the "trending extensions" code and introduce a boolean flag like the one we have for featured themes (I also fixed an edge case related to that, hence the changes in the `versions` and `home` reducers).

### CODE REVIEW

File: tests/unit/amo/sagas/test_home.js [hunk 1]
```diff
@@ -171,7 +172,43 @@ describe(__filename, () => {
       expect(expectedAction).toEqual(loadAction);
     });
 
-    it('does not fetch featured themes if includeFeaturedThemes is false', async () => {
+    it('does not fetch featured themes if includeTrendingExtensions is false', async () => {
```
Comment (bobsilverberg) [inline tests/unit/amo/sagas/test_home.js]: I believe this should be "does not fetch *trending extensions* if includeTrendingExtensions is false".

File: tests/unit/amo/sagas/test_home.js [hunk 2]
```diff
@@ -171,7 +172,43 @@ describe(__filename, () => {
...
+          trendingExtensions: null,
+        },
+      });
+
+      const expectedAction = await sagaTester.waitFor(loadAction.type);
+      mockSearchApi.verify();
+      expect(expectedAction).toEqual(loadAction);
+    });
+
+    it('does not fetch trending extensions if includeFeaturedThemes is false', async () => {
```
Comment (bobsilverberg) [inline tests/unit/amo/sagas/test_home.js]: I believe this should be "does not fetch *featured themes* if includeFeaturedThemes is false".

File: tests/unit/core/reducers/test_versions.js [hunk 3]
```diff
@@ -342,6 +342,20 @@ describe(__filename, () => {
...
+          undefined,
+          loadHomeAddons({
+            collections: [],
+            shelves: {
+              featuredExtensions: null,
+            },
+          }),
+        );
+
+        expect(getVersionById({ state, id: versionId })).toEqual(null);
```
Comment (bobsilverberg) [inline tests/unit/core/reducers/test_versions.js]: Because `versionId` is irrelevant to the test, perhaps this should just check that the state after calling `loadHomeAddons` is still `initialState`?
